specifyweb/frontend/js_src/lib/components/InitialContext/treeRanks.ts

```diff
@@ -3,32 +3,36 @@
  * given discipline.
  */
 
+import { ajax } from '../../utils/ajax';
+import { getCache } from '../../utils/cache';
 import { f } from '../../utils/functools';
 import type { RA } from '../../utils/types';
 import { defined } from '../../utils/types';
-import {
-  caseInsensitiveHash,
-  sortFunction,
-  unCapitalize,
-} from '../../utils/utils';
-import { fetchRelated } from '../DataModel/collection';
+import { caseInsensitiveHash } from '../../utils/utils';
 import type {
   AnySchema,
   AnyTree,
+  SerializedRecord,
   SerializedResource,
 } from '../DataModel/helperTypes';
-import type { SpecifyResource } from '../DataModel/legacyTypes';
+import type { SpecifyResource } from '../DataModel/schema';
 import { fetchContext as fetchDomain, schema } from '../DataModel/schema';
-import { getDomainResource } from '../DataModel/scoping';
 import { serializeResource } from '../DataModel/serializers';
 import { genericTables } from '../DataModel/tables';
-import type { Tables } from '../DataModel/types';
+import type { GeographyTreeDef, Tables } from '../DataModel/types';
 
-let treeDefinitions: {
-  readonly [TREE_NAME in AnyTree['tableName']]: {
-    readonly definition: SpecifyResource<Tables[`${TREE_NAME}TreeDef`]>;
+type TreeInformationBackend = {
+  readonly [TREE_NAME in AnyTree['tableName']]: RA<{
+    readonly definition: SerializedRecord<Tables[`${TREE_NAME}TreeDef`]>;
+    readonly ranks: RA<SerializedRecord<Tables[`${TREE_NAME}TreeDefItem`]>>;
+  }>;
+};
+
+export let treeDefinitions: {
+  readonly [TREE_NAME in AnyTree['tableName']]: RA<{
+    readonly definition: SerializedResource<Tables[`${TREE_NAME}TreeDef`]>;
     readonly ranks: RA<SerializedResource<Tables[`${TREE_NAME}TreeDefItem`]>>;
-  };
+  }>;
 } = undefined!;
```

Comment:
The type looks duplicated? How about
```
let treeDefinition: TreeInformationBackend = undefined;
```
Also, I think the type name `TreeInformationBackend` could be clearer. How about just `TreeInformation` or `TreeDefinitions`